### ISSUE TITLE
Cannot delete property '--' of #<Object> for lerna 3.18.4

### ISSUE THREAD

**Issue by plantain-00** [lerna/lerna#2348]
I exec `lerna exec -- npm view \$LERNA_PACKAGE_NAME`, then an error appears:
```bash
$ lerna -v
3.18.4
$ lerna exec -- npm view \$LERNA_PACKAGE_NAME
/<proj>/node_modules/yargs/yargs.js:1184
      else throw err
           ^

TypeError: Cannot delete property '--' of #<Object>
    at Object.Yargs.self._copyDoubleDash (/<proj>/node_modules/yargs/yargs.js:1196:5)
    at Object.parseArgs [as _parseArgs] (/<proj>/node_modules/yargs/yargs.js:1097:60)
    at Object.parse (/<proj>/node_modules/yargs/yargs.js:578:25)
    at main (/<proj>/node_modules/lerna/index.js:44:6)
    at Object.<anonymous> (/<proj>/node_modules/lerna/cli.js:11:15)
    at Module._compile (internal/modules/cjs/loader.js:956:30)
    at Object.Module._extensions..js (internal/modules/cjs/loader.js:973:10)
    at Module.load (internal/modules/cjs/loader.js:812:32)
    at Function.Module._load (internal/modules/cjs/loader.js:724:14)
    at Module.require (internal/modules/cjs/loader.js:849:19)
```

## Expected Behavior

No error

## Current Behavior

The error

## Possible Solution

Maybe from yargs 14.2.1

## Steps to Reproduce (for bugs)

1. create an empty respository
2. yarn init
3. yarn global add lerna
4. lerna exec -- npm view \$LERNA_PACKAGE_NAME

<details><summary>lerna.json</summary><p>

```json
<!-- Please paste your `lerna.json` here -->
no lerna.json
```
</p></details>

<details><summary>lerna-debug.log</summary><p>

```txt
<!-- If you have a `lerna-debug.log` available, please paste it here -->
<!-- Otherwise, feel free to delete this <details> block -->
no lerna-debug.log
```
</p></details>

## Your Environment

| Executable | Version |
| ---: | :--- |
| `lerna --version` | 3.18.4 |
| `npm --version`  | 6.11.3 |
| `yarn --version` | 1.12.1 |
| `node --version` | v12.12.0 |

| OS | Version |
| --- | --- |
| MacOS Mojave | 10.14.6 |

**Comment by Salakar** [comment]
Ditto, same version here;

```bash
lerna notice cli v3.18.4
lerna info Symlinking packages and binaries
/<proj>/node_modules/@lerna/cli/node_modules/yargs/yargs.js:1184
      else throw err
           ^

TypeError: Cannot delete property '--' of #<Object>
    at Object.Yargs.self._copyDoubleDash (/<proj>/node_modules/@lerna/cli/node_modules/yargs/yargs.js:1196:5)
    at Object.parseArgs [as _parseArgs] (/<proj>/node_modules/@lerna/cli/node_modules/yargs/yargs.js:1097:60)
    at Object.parse (/<proj>/node_modules/@lerna/cli/node_modules/yargs/yargs.js:578:25)
    at main (/<proj>/node_modules/lerna/index.js:44:6)
    at Object.<anonymous> (/<proj>/node_modules/lerna/cli.js:11:15)
    at Module._compile (internal/modules/cjs/loader.js:956:30)
    at Object.Module._extensions..js (internal/modules/cjs/loader.js:973:10)
    at Module.load (internal/modules/cjs/loader.js:812:32)
    at Function.Module._load (internal/modules/cjs/loader.js:724:14)
    at Function.Module.runMain (internal/modules/cjs/loader.js:1025:10)
```

**Comment by moki** [comment]
remove double dash `--` and instead group command with the single quotes `npm view \$LERNA_PACKAGE_NAME` also i believe escaped double quotes should work`\"npm view \$LERNA_PACKAGE_NAME\"`

**Comment by plantain-00** [comment]
The document should be updated if this is intended: https://github.com/lerna/lerna/tree/master/commands/exec#usage
Then it's a break change.

**Comment by arciisine** [comment]
Got hit by the same bug as well.  Looks like it was introduced in `3.18.0` when upgrading to `yargs` v4. I agree this seems like a bug and not intended behavior.

**Comment by brianjquinn** [comment]
I'm getting this with `lerna run` as well when trying to pass args to the underlying npm script with lerna v3.18.4.

`lerna run start --scope some-package -- -c someoption`

results in:
```
<proj>/node_modules/yargs/yargs.js:1184
      else throw err
           ^

TypeError: Cannot delete property '--' of #<Object>
    at Object.Yargs.self._copyDoubleDash (<proj>l/node_modules/yargs/yargs.js:1196:5)
    at Object.parseArgs [as _parseArgs] (<proj>/node_modules/yargs/yargs.js:1097:60)
    at Object.parse (<proj>/node_modules/yargs/yargs.js:578:25)
    at main (<proj>/node_modules/lerna/index.js:44:6)
    at Object.<anonymous> (<proj>/node_modules/lerna/cli.js:11:15)
    at Module._compile (internal/modules/cjs/loader.js:778:30)
    at Object.Module._extensions..js (internal/modules/cjs/loader.js:789:10)
    at Module.load (internal/modules/cjs/loader.js:653:32)
    at tryModuleLoad (internal/modules/cjs/loader.js:593:12)
    at Function.Module._load (internal/modules/cjs/loader.js:585:3)
```

**Comment by moki** [comment]
@arciisine this is the behavior that is covered by tests, one may see with his own eyes by checking out commit link i've provided, but if its seems like a bug to you oh well...

@brianjquinn instead of passing arguments to a sub process after double dash `--`, wrap the whole command with single quotes`lerna run start --scope 'some-package -c someoption'` or `lerna run start --scope \"some-package -c someoption\"`

**Comment by brianjquinn** [comment]
@moki you provided a link to the `lerna version` command. @plantain-00 referred to the `lerna exec` command and I referred to the `lerna run` command, both of which still have handling around "--" args. 

The workaround you provide are absolutely incorrect and do not work. The first example using single quotes wrt to the `--scope` option is a filter and not meant to pass additional arguments to the underlying package and does not work. The second example also does not work and results in "Unknown argument: c".

**Comment by moki** [comment]
> @moki you provided a link to code for the `lerna version` command. @plantain-00 referred to the `lerna exec` command and I referred to the `lerna run` command, both of which still have handling around "--" args. As you acknowledged as well, the "--" args functionality is in the docs.
> 
> The workaround you provide are absolutely incorrect and do not work. The first example using single quotes wrt to the `--scope` option is a filter and not meant to pass additional arguments to the underlying package and does not work. The second example also does not work and results in "Unknown argument: some-package -c someoption".

misread your case anyway still provided fix is relevant to the original problem, also i believe the functionality is shared between commands.

**Comment by brianjquinn** [comment]
> @brianjquinn and if you are looking for a relevant for you copy and paste, given that start is the command you wanna run and some-package is the scope and someoption is the argument to the command: lerna run --scope some-package 'start someoption'

once again, does not work. That would specify you run the command `start someoption` in the package `some-package`.

**Comment by plantain-00** [comment]
Seems introduced from: https://github.com/yargs/yargs/pull/1459

**Comment by ndragon75** [comment]
was fixed in `yargs@v15.0`:
https://github.com/yargs/yargs/commits/v15.0.0

**Comment by Vishesh30** [comment]
Stuck with same issue.! Any workaround.?? Is a patch coming through?

**Comment by samarpanB** [comment]
Same here. can someone sugest a workaround here ? Our builds are failing. Downgrading to version 3.17 also didnt work

**Comment by joeycozza** [comment]
@samarpanB I tried downgrading lerna in my package.json to 3.17.0 , and it DID work for me. This is what that line looks like.
```
    "lerna": "3.17.0",
```

**Comment by sbrichardson** [comment]
This resolved for me also until sorted out.

> @samarpanB I tried downgrading lerna in my package.json to 3.17.0 , and it DID work for me. This is what that line looks like.
> 
> ```
>     "lerna": "3.17.0",
> ```

**Comment by Robinnnnn** [comment]
> Downgrading to version 3.17 also didnt work

@samarpanB make sure you correctly pin the version in your `package.json` as `"3.17.0"` and not `"^3.17.0"`.

also make sure to clear out your `node_modules` and re-run a clean `npm install` (or `yarn`, if that's what you use)

**Comment by Vishesh30** [comment]
Thanks @Robinnnnn Downgrading to 3.17.0 and pinning the same in package.json solves the issue for me

**Comment by nicolo-ribaudo** [comment]
This should be fixed in yargs 14.2.2 (https://github.com/babel/babel/pull/10733#discussion_r347712415)

**Comment by daern91** [comment]
@nicolo-ribaudo it still installs an old version for me, could you please update the lock-file? https://github.com/lerna/lerna/blob/master/package-lock.json#L9522-L9523

**Comment by daern91** [comment]
Added a PR for it here, https://github.com/lerna/lerna/pull/2355

**Comment by evocateur** [comment]
Fixed in v3.18.5
